File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
   int maxSubArray(vector<int>& nums) {
    int sum = 0;
    int prevMaxSum = INT_MIN;
    // vector<int> sums;
    for (int i = 0; i < nums.size(); i++) {
        if(nums[i] > sum+nums[i]) {
            sum = nums[i];
        }else {
            sum += nums[i];
        }
        if(sum > prevMaxSum) {
            prevMaxSum = sum;
        }
    }

    return prevMaxSum;

}
};

File: 53-maximum-subarray/maximum-subarray.ts
function maxSubArray(nums: number[]): number {
    let prevMax: number = Number.MIN_SAFE_INTEGER;
    let sum: number = 0;
    for (let i of nums) {
        if (i > i + sum) {
            sum = i;
        }else {
            sum += i;
        }
        if (sum > prevMax) {
            prevMax = sum;
        }

    }

    return prevMax
};